ckan/controllers/user.py

```diff
@@ -162,10 +166,90 @@ def register(self, data=None, errors=None, error_summary=None):
 
         return self.new(data, errors, error_summary)
 
+    def _generate_internal_logs(self, log_message):
+        '''
+        This helper function is called if sending a new user registration email fails.
+        '''
+
+        logging.warning("There was an error sending the email. Writing to log file.")
+
+        os.chdir(os.environ['REGISTRATION_ERROR_LOG_FILE_PATH'])
+
+        with open(os.environ['REGISTRATION_ERROR_LOG_FILE_NAME'], "a") as fp:
+            try:
+                fp.write(log_message)
+                fp.write("\n\n")
+            except:
+                logging.warning("There is an error writing to the log file. Dumping the output to console.")
+                logging.warning(log_message)
+
+        logging.warning("Written to log file.")
+
+    def email_new_user_details_using_mailgun(self, request_data):
+        request_params = dict(request_data)
+
+        email_body = "There is a new user registration request. \
+        \nThe user's details are as follows: \
+        \n\
+        \nUsername: {0}\
+        \nName: {1} \
+        \nEmail: {2} \
+        \nReason for Request: {3} \
+        \nProject of Interest: {4} ".format(request_params['name'], request_params['fullname'], request_params['email'], request_params['request_reason'], request_params['project_of_interest'])
```

Comment:
minor style point - you don't need to number to arguments. personally I prefer the named keyword argument usage of `format()` if there a lot of arguments, then ordering no longer matters

Reply:
Fixed.